clients/python/test/test_osparc/test_basic.py

```diff
@@ -87,3 +93,42 @@ def test_parent_project_validation(faker, valid: bool):
         os.environ["OSPARC_NODE_ID"] = f"{faker.text()}"
         with pytest.raises(pydantic.ValidationError):
             _ = osparc._settings.ParentProjectInfo()
+
+
+def test_pagination_iterator(
+    faker: Faker, page_file: osparc.PageFile, api_client: osparc.ApiClient
+):
+    next_page_url = urlparse(page_file.links.next)
+    _base_url = f"{next_page_url.scheme}://{next_page_url.netloc}"
+    _auth = httpx.BasicAuth(
+        username=api_client.configuration.username,
+        password=api_client.configuration.password,
+    )
+
+    def _sideeffect(all_items: List, request: httpx.Request):
+        all_items += page_file.items
+        if len(all_items) >= page_file.total:
+            all_items = all_items[: page_file.total]
+            page_file.links.next = None
+        return httpx.Response(status_code=200, json=page_file.to_dict())
+
+    with respx.mock(
+        base_url=_base_url,
+        assert_all_called=True,
+    ) as respx_mock:
+        server_items: List[osparc.File] = page_file.items
+        respx_mock.get(urlparse(page_file.links.next).path).mock(
+            side_effect=partial(_sideeffect, server_items)
+        )
+
+        pagination_iterator = PaginationIterator(
+            lambda: page_file, api_client=api_client, base_url=_base_url, auth=_auth
+        )
+        client_items = [item for item in pagination_iterator]
+        assert len(server_items) > 0
+        assert all(si == ci for si, ci in zip(server_items, client_items))
+
+        first_client_item = next(pagination_iterator)
+        assert first_client_item == server_items[0]
+
+        assert len(pagination_iterator) == page_file.total
```

Comment:
?? this i am not sure i like. I would not expect that an iterator can be measured ...